src/charm.py

```diff
@@ -157,20 +161,44 @@ def _is_image_relation_ready_set_status(self, cloud_config: state.CloudConfig) -
             return False
         return True
 
-    def _run(self) -> None:
+    def _run(self, cloud_id: str | None = None) -> None:
         """Trigger an image build.
 
         This method requires that the clouds.yaml are properly installed with build cloud and
         upload cloud authentication parameters.
+
+        Args:
+            cloud_id: The cloud ID to upload the image to. If None, the image will be uploaded to
+                all clouds.
         """
+        start_ts = time.time()
         self.unit.status = ops.ActiveStatus("Building image.")
+        logger.info(
+            "Building image and uploading to %s.", (cloud_id if cloud_id else "all clouds")
+        )
         builder_config = state.BuilderConfig.from_charm(self)
+        config_matrix = self._get_configuration_matrix(builder_config=builder_config)
+        static_config = self._get_static_config(builder_config=builder_config)
+        if cloud_id:
+            static_config.cloud_config.upload_clouds = [cloud_id]
```

Comment:
So not supplying any `cloud_id` would upload to all clouds already?

Reply:
From `app/src/github_runner_image_builder/openstack_builder.py` line 246, it seems setting to none will upload to the same cloud.

Is the comment at line 171 correct?

Reply:
`static_config.cloud_config.upload_clouds` contains all clouds where to upload the image. If cloud_id is specified, this is overriden to be a single cloud. Otherwise we keep the value as it is. 

I think the comment in `app/src/github_runner_image_builder/openstack_builder.py line 246` is wrong